### ISSUE TITLE
TargetRecordingGetHandler should use Vertx async io

### ISSUE THREAD

**Issue by andrewazores** [cryostatio/cryostat#568]
https://github.com/cryostatio/cryostat/blob/74328d3268156f001b8977ab769601da7a3e2fd5/src/main/java/io/cryostat/net/web/http/api/v1/TargetRecordingGetHandler.java#L134

**Comment by hareetd** [comment]
I read the Vert.x docs for Streams and had a couple of questions. 

1. From what I can tell, in order to transfer the data using `Pump` the `Object` being read from needs to implement `ReadStream` while the object being written to needs to implement `WriteStream`.  Currently, the data flow is: 

`InputStream` -> `byte[]` -> `Buffer` -> `HttpServerResponse`

Since `HttpServerResponse` implements `WriteStream`, we're good on that end, but the problem is that we don't (currently) have a `ReadStream` to work with. From what I've seen online, there are custom implementations that wrap `InputStream` inside a class extending `ReadStream` but I think that isn't the right approach. Any suggestions?

2. In its description, the `ctx.response.write()` method states "to avoid running out of memory by putting too much on the write queue, check the writeQueueFull method before writing. This is done automatically if using a Pump". I'm assuming this is the reason behind opening this issue?

**Comment by andrewazores** [comment]
> Second, in its description, the ctx.response.write() method states "to avoid running out of memory by putting too much on the write queue, check the writeQueueFull method before writing. This is done automatically if using a Pump". I'm assuming this is the reason behind opening this issue?

Pretty much. It would also just be nice from a code cleanliness point of view to not mix and match I/O from different frameworks (JDK standard library vs Vert.x).

We're stuck awkwardly in the middle between Vert.x on our API side making us use Vert.x I/O, and on our -core side because of our JMC dependencies using standard JDK I/O (`java.io` and `java.nio` packages). Ideally we would just pick one or the other. Barring that, we try to maintain a clean separate, and wherever necessary we implement some glue code to stick things together.

> From what I've seen online, there are custom implementations that wrap InputStream inside a class extending ReadStream but I think that isn't the right approach. Any suggestions?

That might just be the kind of glue code we need, but doing it the most naive way like what I wrote in that handler's current state is probably missing out on some of the async I/O performance benefits that Vert.x can offer. At worst it would be nice to refactor that into a utility class that implements `ReadStream` for better readability and maintainability.

https://stackoverflow.com/questions/66158455/how-to-create-a-vert-x-readstream-from-an-inputstream

The `vertx-java.io` library mentioned in this post (right after I filed this issue :-) ) looks pretty good, but I doubt that's an available downstream dependency we can pull in. Thankfully it's MIT licensed, so I think we should be able to grab a copy of the source file we need and stick it into our source tree directly.

This gist - a fork of the one comment on the SO post - looks pretty similar too.

https://gist.github.com/TheAntimist/af95e3ce63682e1d531b8a0b5a649fa5

I haven't read too deeply into either to evaluate them or judge which is better.
